native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractStatementParameterProcessor.java

```diff
@@ -197,81 +197,88 @@ protected abstract int setCustomBOpenRecord(Connection connection, PreparedState
+                    setIntegerArray(connection, preparedStatement, index, objectArray);
+                } else if (type.equals(Constants.SqlTypes.BOOLEAN_TYPE) ||
+                        type.equals(Constants.SqlTypes.OPTIONAL_BOOLEAN)) {
+                    setBooleanArray(connection, preparedStatement, index, objectArray);
+                } else if (type.equals(Constants.SqlTypes.FLOAT_TYPE) ||
+                        type.equals(Constants.SqlTypes.OPTIONAL_FLOAT)) {
+                    setFloatArray(connection, preparedStatement, index, objectArray);
+                } else if (type.equals(Constants.SqlTypes.DECIMAL_TYPE) ||
+                        type.equals(Constants.SqlTypes.OPTIONAL_DECIMAL)) {
+                    setDecimalArray(connection, preparedStatement, index, objectArray);
+                } else {
+                    // Cannot be reached as this is validated with `ArrayValueType` in ballerina
+                    throw new UnsupportedTypeError(type + "array type", index);
+                }
+                return Types.ARRAY;
+            } else if (object instanceof BObject) {
+                BObject objectValue = (BObject) object;
+                setSqlTypedParam(connection, preparedStatement, index, objectValue);
+                if (returnType) {
+                    return getSQLType(objectValue);
+                }
+                return 0;
+            } else if (object instanceof BXml) {
+                setXml(connection, preparedStatement, index, (BXml) object);
+                return Types.SQLXML;
+            } else if (object instanceof BMap) {
+                return setBMapParams(connection, preparedStatement, index, (BMap) object, returnType);
             } else {
-                // Cannot be reached as this is validated with `ArrayValueType` in ballerina
-                throw new UnsupportedTypeError(type + "array type", index);
-            }
-            return Types.ARRAY;
-        } else if (object instanceof BObject) {
-            BObject objectValue = (BObject) object;
-            setSqlTypedParam(connection, preparedStatement, index, objectValue);
-            if (returnType) {
-                return getSQLType(objectValue);
+                // Cannot be achieved since this is validated at compiler for `Value`
+                throw new UnsupportedTypeError(object.getClass().getName(), index);
             }
-            return 0;
-        } else if (object instanceof BXml) {
-            setXml(connection, preparedStatement, index, (BXml) object);
-            return Types.SQLXML;
-        } else if (object instanceof BMap) {
-            return setBMapParams(connection, preparedStatement, index, (BMap) object, returnType);
-        } else {
-            // Cannot be achieved since this is validated at compiler for `Value`
-            throw new UnsupportedTypeError(object.getClass().getName(), index);
+        } catch (SQLException e) {
```

Comment:
We cant covert all SQLexceptions as DataError. Only exceptions pertaining to data access needs to be changed to DataError.